File: on-boarding/leetcode/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int j=0;
        int sum=0;
        int minSize=100000;
        boolean set=false;
        for(int i=0;i<nums.length;i++){
            sum+=nums[i];
            while(sum>=target){
                minSize=Math.min(minSize,(i-j)+1);
                sum -= nums[j];
                j++;
                set=true;
            }
        }
        if(!set){
            minSize=0;
        }
        return minSize;
    }
}

File: on-boarding/minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        min_ = float('inf')
        running_sum = 0
        left = 0

        for right in range(len(nums)):
            running_sum += nums[right]

            while running_sum >= target:
                min_ = min(min_,right-left+1)
                running_sum -= nums[left]
                left += 1
        
        return min_ if min_ != float('inf') else 0